src/UI/WB.UI.Designer/Code/QuestionnaireHelper.cs

```diff
@@ -182,7 +184,7 @@ private QuestionnaireListViewModel GetFolder(QuestionnaireListViewFolder x, bool
             var questionnaireDocument = questionnaireView.Source;
             
             var maxSequenceByQuestionnaire = this.questionnaireChangeItemStorage.QuestionnaireChangeRecords
-                .Where(y => y.QuestionnaireId == id.FormatGuid())
+                .Where(y => y.QuestionnaireId == questionnaireView.PublicKey.FormatGuid())
                 .Select(y => (int?)y.Sequence)
                 .Max();
             

```

Comment:
you should set revision field from QuestionnaireRevision, becouse current code set last revision always

Reply:
Add changes to load all data but single revision